### PR TITLE
Add new features

### PR DESCRIPTION
**Pull request by Dec1s1on**
360-degree depth map inference from a monocular image

### CODE REVIEW

File: modules/depth_prediction/inference.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import matplotlib as mpl
+import matplotlib.cm as cm
+
+import torch
+from torchvision import transforms, datasets
+
+from openvino.inference_engine import IECore
+
+
+image_path = "C:/Users/Avenger/Desktop/gsoc/monodepth2/assets/test_image.jpg"
```
Comment (zongwave) [inline modules/depth_prediction/inference.py:16]: we should avoid using absolute file paths.

File: modules/depth_prediction/layers.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
+# Copyright Niantic 2019. Patent Pending. All rights reserved.
```
Comment (zongwave) [inline modules/depth_prediction/layers.py:1]: is it necessary to include all the Monodepth2 training source code in this patch? can you just add a pre-trained model in this patch?

File: modules/depth_prediction/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
+This is the a depth estimation models using the method described in
```
Comment (zongwave) [inline modules/depth_prediction/README.md:1]: please remove all the files under path libxcam/modules/dnn

Comment (zongwave) [reply modules/depth_prediction/README.md:1]: please remove python cache files

**Review by zongwave — CHANGES_REQUESTED** [review]

File: modules/depth_prediction/layers.py (hunk 2 above)
Comment (Dec1s1on) [reply modules/depth_prediction/layers.py:1]: I have prepared the pre trained model, and the user can download the model to the specified folder and reasoning directly. The inference part does not need this part of code.

This part of the code about the network is introduced for the model transformation part. Considering that if multiple .PTH file format models trained by users want to be converted to IR format, the network needs to be loaded first for analysis and integrated into an intermediate file in .Onnx format.

Comment (zongwave) [reply modules/depth_prediction/layers.py:1]: You can add comments in README.md to introduce how to train a Monodepth2 model from the original github repo.
The patch should only include inference part, add some introduction of the external model being used.
Please refer to an example of openvino model zoo
https://github.com/openvinotoolkit/open_model_zoo/tree/master/models/public/fcrn-dp-nyu-depth-v2-tf

Comment (zongwave) [reply modules/depth_prediction/layers.py:1]: Also please remove all python cache files
modules/depth_prediction/__pycache__/layers.cpython-36.pyc
modules/depth_prediction/__pycache__/utils.cpython-36.pyc

**Review by zongwave — CHANGES_REQUESTED** [review]

**Pull request closed without merging**